File: Backend/SEVestTourAPI/Entities/StyleOption.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace SEVestTourAPI.Entities;

[Table("StyleOption")]
public partial class StyleOption
{
    [Key]
    [Column("StyleOptionID")]
    public int StyleOptionId { get; set; }

    [Column("StyleID")]
    public int? StyleId { get; set; }

    [StringLength(100)]
    public string? OptionType { get; set; }

    [StringLength(100)]
    public string? OptionValue { get; set; }
    public decimal? Price { get; set; }
    [ForeignKey("StyleId")]
    [InverseProperty("StyleOptions")]
    public virtual Style? Style { get; set; }
}

File: Mobile/mobileapp/lib/models/styleOption.dart
class StyleOption {
  int? styleOptionId;
  int? styleId;
  String? optionType;
  String? optionValue;
  double? price;

  StyleOption(
      {this.styleOptionId,
        this.styleId,
        this.optionType,
        this.optionValue,
        this.price});

  StyleOption.fromJson(Map<String, dynamic> json) {
    styleOptionId = json['styleOptionId'];
    styleId = json['styleId'];
    optionType = json['optionType'];
    optionValue = json['optionValue'];
    price = json['price'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['styleOptionId'] = this.styleOptionId;
    data['styleId'] = this.styleId;
    data['optionType'] = this.optionType;
    data['optionValue'] = this.optionValue;
    data['price'] = this.price;
    return data;
  }
}
